src/main/java/org/apache/maven/plugins/javadoc/AbstractJavadocMojo.java

```diff
@@ -4414,9 +4430,15 @@ private void addCommandLineArgFile( Commandline cmd, File javadocOutputDirectory
             cmd.createArg().setValue( "@" + FILES_FILE_NAME );
         }
 
+        /* default to platform encoding */
+        String encoding = null;
+        if ( JAVA_VERSION.compareTo( SINCE_JAVADOC_9 ) >= 0 )

```

Comment:
Please switch to `isAtLeast()` or `isBefore()`, which are better readable.

Reply:
Don't forget to squash your commits

Reply:
I addressed your points and squashed the commits.

Ran mvn -Prun-its verify with JDK8 and JDK9 on Windows successfully.